libglusterfs/src/syscall.c

```diff
@@ -187,7 +187,9 @@ sys_readdir(DIR *dir, struct dirent *de)
 ssize_t
 sys_readlink(const char *path, char *buf, size_t bufsiz)
 {
-    return FS_RET_CHECK(readlink(path, buf, bufsiz), errno);
+    int sz = readlink(path, buf, bufsiz);
```

Comment:
This is a simple wrapper of the `readlink()` system call. The behavior of this function should be exactly the same as calling `readlink()` directly. What should need to be updated are the callers of this function.

Reply:
Thanks for the suggestion. I have made the required changes and have added the null character at the caller of the function.